src/components/ui/SlideOver/SlideOver.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState, useCallback } from 'react'
+import type { ReactNode, HTMLAttributes } from 'react'
+import { Modal as UIModal } from '@faststore/ui'
+
+import './slide-over.scss'
+
+type Direction = 'leftSide' | 'rightSide'
+type WidthSize = 'full' | 'partial'
+type FadeType = 'in' | 'out'
+
+interface SlideOverProps extends HTMLAttributes<HTMLDivElement> {
+  isOpen: boolean
+  direction: Direction
+  size: WidthSize
+  children: ReactNode
+  /**
+   * This function is called whenever the user clicks outside
+   * the modal content
+   */
+  onDismiss?: () => void
+  /**
+   * This callback function enables the transition effect on children "close" button
+   */
+  onDismissTransition: (callback: () => unknown) => unknown
+}
+
+const SlideOver = ({
+  isOpen,
+  onDismiss,
+  onDismissTransition,
+  direction = 'leftSide',
+  size = 'full',
+  children,
+  ...otherProps
+}: SlideOverProps) => {
+  const [fadeType, setFadeType] = useState<FadeType>()
+
+  const handleClose = useCallback(() => {
+    setFadeType('out')
+  }, [])
+
+  useEffect(() => {
+    if (isOpen) {
+      setFadeType('in')
+    }
+  }, [isOpen])
+
+  useEffect(() => {
+    if (handleClose) {
```

Comment:
I'm not sure if we need this `if` here since `handleClose` is a `function` and is always `true`. 👀

Reply:
I think you are right! 🙈